api/client/kubewaitingcontainer/kubewaitingcontainer.go

```diff
@@ -0,0 +1,166 @@
+// Copyright 2023 Gravitational, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package kubewaitingcontainer
+
+import (
+	"context"
+
+	"github.com/gravitational/trace"
+
+	kubewaitingcontainerpb "github.com/gravitational/teleport/api/gen/proto/go/teleport/kubewaitingcontainer/v1"
+	"github.com/gravitational/teleport/api/types/kubewaitingcontainer"
+	convert "github.com/gravitational/teleport/api/types/kubewaitingcontainer/convert/v1"
+)
+
+// Client is a KubeWaitingContainers client.
+type Client struct {
+	grpcClient kubewaitingcontainerpb.KubeWaitingContainersServiceClient
+}
+
+// NewClient creates a new KubeWaitingContainers client.
+func NewClient(grpcClient kubewaitingcontainerpb.KubeWaitingContainersServiceClient) *Client {
+	return &Client{
+		grpcClient: grpcClient,
+	}
+}
+
+// KubeWaitingContainerRequest is a request for a specific Kubernetes
+// ephemeral container that is waiting to be created until moderated
+// session conditions are met.
+type KubeWaitingContainerRequest struct {
+	// Username is the Teleport user that attempted to create the container
+	Username string `json:"username" yaml:"username"`
+	// Cluster is the Kubernetes cluster of the container
+	Cluster string `json:"cluster" yaml:"cluster"`
+	// Namespace is the Kubernetes namespace of the container
+	Namespace string `json:"namespace" yaml:"namespace"`
+	// PodName is the parent pod of the container
+	PodName string `json:"pod_name" yaml:"pod_name"`
+	// ContainerName is the name of the container
+	ContainerName string `json:"container_name" yaml:"container_name"`
+}
```

Comment:
See my reply on your first comment.

Reply:
That's a good point

Reply:
There is a bit of code duplication here, but requiring the caller to pass in the proto definition implies that all fields are required to be properly set which isn't true. The metadata and the `Spec.Patch` wouldn't be needed for example. I think keeping this struct makes the client easier to use.

Reply:
Ah, I thought you meant the `KubernetesWaitingContainer` message should be used instead of `KubeWaitingContainerRequest`. I removed struct and used the appropriate proto messages instead.